File: 백준/Gold/1083. 소트/소트.java
import java.io.*;
import java.util.StringTokenizer;
import java.util.ArrayList;
import java.util.Arrays;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[] nums = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }
        ArrayList<Integer> sortedNum = new ArrayList<>();
        int S = Integer.parseInt(br.readLine());
        int cnt = 0;
        while (S > 0) {
            if (sortedNum.size() == N) {
                break;
            }
            if (S == 1) {
                for (int i = cnt; i < N-1; i++) {
                    if (nums[i] < nums[i+1]) {
                        int tmp = nums[i];
                        nums[i] = nums[i+1];
                        nums[i+1] = tmp;
                        break;
                    }
                }
                break;
            }
            int[][] numsIdx = new int[N-cnt][2];
            for (int i = cnt; i < N; i++) {
                numsIdx[i-cnt][0] = nums[i];
                numsIdx[i-cnt][1] = i;
            }
            Arrays.sort(numsIdx, (o1, o2 )-> o2[0] - o1[0]);
            for (int[] num : numsIdx) {
                if (num[1] - cnt <= S) {
                    for (int j = num[1]; j > cnt; j--) {
                        int tmp = nums[j];
                        nums[j] = nums[j-1];
                        nums[j-1] = tmp;
                    }
                    S -= num[1] - cnt;
                    sortedNum.add(num[0]);
                    break;
                }
            }
            cnt++;
        }
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < N; i++) {
            sb.append(nums[i]).append(" ");
        }
        System.out.println(sb);
    }
}

File: 백준/Gold/1083. 소트/소트.py
N = int(input())
nums = list(map(int, input().split()))
S = int(input())

sorted_num = []
cnt = 0
while S > 0:
    if nums == sorted_num:
        break
    if S == 1:
        for i in range(cnt, N-1):
            if nums[i] < nums[i+1]:
                nums[i], nums[i + 1] = nums[i + 1], nums[i]
                break
        break
    nums_idx = []
    for i in range(cnt, N):
        if nums[i] not in sorted_num:
            nums_idx.append((nums[i], i))
    nums_idx.sort(key=lambda x: x[0], reverse=True)
    for i in range(len(nums_idx)):
        if nums_idx[i][1] - cnt <= S:
            for j in range(nums_idx[i][1], cnt, -1):
                nums[j], nums[j-1] = nums[j-1], nums[j]
            S -= nums_idx[i][1] - cnt
            sorted_num.append(nums_idx[i][0])
            break
    cnt += 1
print(*nums)